src/OpenSpaceToolkit/Physics/Environment/Ephemerides/SPICE/Engine.cpp

```diff
@@ -228,16 +231,41 @@ Engine::Mode Engine::DefaultMode()
 
 Array<Kernel> Engine::DefaultKernels(const Directory& aLocalRepository)
 {
-    static const Array<Kernel> defaultKernels = {
+    // Use regex to pull Earth body shape, orientation and leap second kernels, as the file name is often updated.
+    using iterator = std::filesystem::directory_iterator;
 
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("naif0012.tls"))),  // Leap seconds
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("de430.bsp"))),     // Ephemeris
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("pck00011.tpc"))
-        ),  // System body shape and orientation constants
+    auto findKernel = [&aLocalRepository](const String aRegexString) -> Kernel
```

Comment:
Why not making this a public method of the Manager class?

Reply:
At this point it probably makes sense, huh? This quick fix MR has accidentally morphed into doing it the right way 😆